dataprocspawner/customize_cluster.py

```diff
@@ -57,240 +145,787 @@ def get_custom_cluster_html_form(autoscaling_policies, node_types):
-    <div class="form-group">
-      <label for="custom_labels">User defined labels</label>
-      <input name="custom_labels" class="form-control" placeholder="key1:value1,key2:value2"
-        value=''>
-      </input>
-    </div>"""
+    <label class="mdc-text-field jupyter-text-field mdc-text-field--outlined">
+      <span class="mdc-notched-outline">
+        <span class="mdc-notched-outline__leading"></span>
+        <span class="mdc-notched-outline__notch">
+          <span class="mdc-floating-label" id="custom_labels">
+            User defined labels
+          </span>
+        </span>
+        <span class="mdc-notched-outline__trailing"></span>
+      </span>
+      <input
+        type="text"
+        class="mdc-text-field__input"
+        name="custom_labels"
+        aria-labelledby="custom_labels"/>
+    </label>
+    <div class="mdc-text-field-helper-line">
+      <div class="mdc-text-field-helper-text" aria-hidden="true">
+        Example: 'key1:value1,key2:value2'
+      </div>
+    </div>
+    <br />"""
 
   html_init_actions = """
-    <div class="form-group">
-      <label for="init_actions">Initialization actions</label>
-      <input name="init_actions" class="form-control"
-        placeholder="gs://<init_action1>,gs://<init_action2>" value="">
-      </input>
-    </div>"""
+    <label class="mdc-text-field jupyter-text-field mdc-text-field--outlined">
+      <span class="mdc-notched-outline">
+        <span class="mdc-notched-outline__leading"></span>
+        <span class="mdc-notched-outline__notch">
+          <span class="mdc-floating-label" id="init_actions">
+            Initialization actions
+          </span>
+        </span>
+        <span class="mdc-notched-outline__trailing"></span>
+      </span>
+      <input
+        type="text"
+        class="mdc-text-field__input"
+        name="init_actions"
+        aria-labelledby="init_actions"/>
+    </label>
+    <div class="mdc-text-field-helper-line">
+      <div class="mdc-text-field-helper-text" aria-hidden="true">
+        Example: 'gs://init-action1,gs://init-action2'
+      </div>
+    </div>
+    <br />"""
 
   html_hive_settings = """
-    <div class="form-group">
-      <label for="hive_host">Hive Metastore host</label>
-      <input name="hive_host" class="form-control" placeholder='' value=''></input>
-    </div>
-    <div class="form-group">
-      <label for="hive_db">Hive Metastore database</label>
-      <input name="hive_db" class="form-control" placeholder='' value=''></input>
-    </div>
-    <div class="form-group">
-      <label for="hive_user">Hive Metastore user name</label>
-      <input name="hive_user" class="form-control" placeholder='' value=''></input>
-    </div>
-    <div class="form-group">
-      <label for="hive_passwd">Hive Metastore password</label>
-      <input name="hive_passwd" class="form-control" placeholder='' value=''></input>
-    </div>"""
+    <label class="mdc-text-field jupyter-text-field mdc-text-field--outlined">
```

Comment:
All of the input labels are missing for the hive related inputs.

Also, would it be possible to group these fields under a "Hive Metastore" group like we do for the node configs?

Reply:
See the comment about the labels above.
Hive settings are now grouped.